trunk/src/dynamic-content/dynamic-content-edit.js

```diff
@@ -0,0 +1,126 @@
+import {__} from '@wordpress/i18n';
+import {Component} from '@wordpress/element';
+import {InspectorControls} from '@wordpress/block-editor';
+import ServerSideRender from '@wordpress/server-side-render';
+
+import '../editor.scss';
+
+class DynamicContentEdit extends Component {
+
+    constructor() {
+        super(...arguments);
+        this.state = {
+            dropdownItems: [],
+            dcItems: [],
+            showDropdown: false,
+            interactive: false,
+        };
+
+        // This binding is necessary to make `this` work in the functions
+        this.render = this.render.bind(this);
+        this.hideOverlay = this.hideOverlay.bind(this);
+    }
+
+    componentDidMount() {
+
+        // We are using the pre-existing PHP Ajax function to allow us to obtain the user's dynamic content.
+        // We need to parse the return value, as the response comes backs as HTML (<select><option>...).
+        let xhr = new XMLHttpRequest()
+
+        xhr.addEventListener('load', () => {
+            let shortcodes = [...xhr.responseText.matchAll(/\[.*?id=&quot;(.*?)&quot; default=&quot;(.*?)&quot;]">(.*?)</g)]
+            let dropdownItems = [];
+            let dcItems = []
+            for (let i = 0; i < shortcodes.length; i++) {
+                dropdownItems.push({dynamicContent_id: shortcodes[i][1], title: shortcodes[i][3]})
+                dcItems[shortcodes[i][1]] = shortcodes[i][2];
+            }
+
+            if (shortcodes.length > 0) {
+                this.setState({
+                    dropdownItems: dropdownItems,
+                    showDropdown: true,
+                    dcItems: dcItems,
+                });
+            }
+
+        })
+
+        xhr.open('POST', ajaxurl)
+        let formData = new FormData();
+        formData.append("action", "get_pardot_dynamicContent_shortcode_select_html");
+        xhr.send(formData);
```

Comment:
do we have to close this request after open/send?

Reply:
Initially I thought it would be close after send but seems I can manually abort

Reply:
yeah I would close it after the data is sent, just to be clean